tools/pipeline-generator/Azure.Sdk.Tools.PipelineGenerator/PipelineGenerationContext.cs

```diff
@@ -68,12 +75,23 @@ private int[] ParseIntArray(string[] strs)
 
         private VssConnection cachedConnection;
 
-        private VssConnection GetConnection()
+        private async Task<VssConnection> GetConnectionAsync()
         {
             if (cachedConnection == null)
             {
-                var pat = Environment.GetEnvironmentVariable(patvar);
-                var credentials = new VssBasicCredential("nobody", pat);
+                VssCredentials credentials;
+                if (string.IsNullOrWhiteSpace(patvar))
+                {
+                    var azureTokenProvider = new AzureServiceTokenProvider();
```

Comment:
Is this using the login information from the local user?

Reply:
it is.   It's MSI login, so it'll use the token endpoint or az cli to get the token if available

Reply:
I was unable to generate a PAT with the scopes required.   When querying for the github repo via ADO REST, I received a 401 no matter the scopes in the token.  It allows for PATless development, then PAT or managed identity for prod.